spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/MessageConverterConfigurer.java

```diff
@@ -81,23 +82,32 @@ public void afterPropertiesSet() throws Exception {
 		Assert.notNull(this.beanFactory, "Bean factory cannot be empty");
 	}
 
+	@Override
+	public void configureInputChannel(MessageChannel messageChannel, String channelName) {
+		configureMessageChannel(messageChannel, channelName, true);
+	}
+
+	@Override
+	public void configureOutputChannel(MessageChannel messageChannel, String channelName) {
+		configureMessageChannel(messageChannel, channelName, false);
+	}
+
 	/**
 	 * Setup data-type and message converters for the given message channel.
 	 * @param channel     message channel to set the data-type and message converters
 	 * @param channelName the channel name
 	 */
-	@Override
-	public void configureMessageChannel(MessageChannel channel, String channelName) {
+	private void configureMessageChannel(MessageChannel channel, String channelName, boolean input) {
 		Assert.isAssignable(AbstractMessageChannel.class, channel.getClass());
 		AbstractMessageChannel messageChannel = (AbstractMessageChannel) channel;
 		final BindingProperties bindingProperties = this.channelBindingServiceProperties.getBindingProperties(
 				channelName);
 		final String contentType = bindingProperties.getContentType();
-		if (bindingProperties.getProducer() != null && bindingProperties.getProducer().isPartitioned()) {
+		if (!input && bindingProperties.getProducer() != null && bindingProperties.getProducer().isPartitioned()) {
```

Comment:
No, it's redundant, but I believe it helps focusing on the different code paths. Eventually we might want to refactor this further, I believe.